src/libraries/System.Net.Security/src/System/Net/Security/SslConnectionInfo.Unix.cs

```diff
@@ -12,14 +17,404 @@ internal partial struct SslConnectionInfo
         private void MapCipherSuite(TlsCipherSuite cipherSuite)
         {
             TlsCipherSuite = cipherSuite;
-
-            TlsCipherSuiteData data = TlsCipherSuiteData.GetCipherSuiteData(cipherSuite);
-            KeyExchangeAlg = (int)data.KeyExchangeAlgorithm;
             KeyExchKeySize = 0;
-            DataCipherAlg = (int)data.CipherAlgorithm;
-            DataKeySize = data.CipherAlgorithmStrength;
-            DataHashAlg = (int)data.MACAlgorithm;
-            DataHashKeySize = data.MACAlgorithmStrength;
+
+            ushort data = GetPackedData(cipherSuite);
+            Debug.Assert(data != 0, $"No mapping found for cipherSuite {cipherSuite}");
+
+            KeyExchangeAlg = (int)s_exchangeAlgorithmTypes[(data >> (16 - (4 * 1)) & 0xF)];
+            DataCipherAlg = (int)s_cipherEnumValues[(data >> (16 - (4 * 2)) & 0xF)];
+            DataKeySize = (int)s_cipherStrengthEnumValues[(data >> (16 - (4 * 3)) & 0xF)];
+            DataHashAlg = (int)s_hashEnumValues[(data >> (16 - (4 * 4)) & 0xF)];
+            DataHashKeySize = GetHashSize((HashAlgorithmType)DataHashAlg);
+
+            static int GetHashSize(HashAlgorithmType hash)
+            {
+                switch (hash)
+                {
+                    case HashAlgorithmType.None:
+                        return 0;
+                    case HashAlgorithmType.Md5:
+                        return 128;
+                    case HashAlgorithmType.Sha1:
+                        return 160;
+                    case HashAlgorithmType.Sha256:
+                        return 256;
+                    case HashAlgorithmType.Sha384:
+                        return 384;
+                    case HashAlgorithmType.Sha512:
+                        return 512;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(hash));
+                }
+            }
+        }
+
+        private static ReadOnlySpan<int> s_exchangeAlgorithmTypes =>
+            new[] { (int)ExchangeAlgorithmType.None, (int)ExchangeAlgorithmType.RsaSign, (int)ExchangeAlgorithmType.RsaKeyX, (int)ExchangeAlgorithmType.DiffieHellman, };
+
+        private static ReadOnlySpan<int> s_cipherEnumValues =>
+            new[] { (int)CipherAlgorithmType.None, (int)CipherAlgorithmType.Null, (int)CipherAlgorithmType.Des, (int)CipherAlgorithmType.Rc2, (int)CipherAlgorithmType.TripleDes, (int)CipherAlgorithmType.Aes128, (int)CipherAlgorithmType.Aes192, (int)CipherAlgorithmType.Aes256, (int)CipherAlgorithmType.Aes, (int)CipherAlgorithmType.Rc4, };
+
+        private static ReadOnlySpan<int> s_cipherStrengthEnumValues =>
+            new[] { (int)CipherAlgorithmStrength.Zero, (int)CipherAlgorithmStrength.Forty, (int)CipherAlgorithmStrength.FiftySix, (int)CipherAlgorithmStrength.OneTwentyEight, (int)CipherAlgorithmStrength.OneSixtyEight, (int)CipherAlgorithmStrength.TwoFiftySix, };
```

Comment:
`CipherAlgorithmStrength` is not really necessary. Can this contain the numbers directly like `new[] { 0, 56, ... }`?

Reply:
Done, also removed the additional enum.